osf_tests/test_node.py

```diff
@@ -3888,6 +3904,36 @@ def test_update_share_correctly_for_projects(self, requests, node, user, request
             graph = kwargs['json']['data']['attributes']['data']['@graph']
             assert graph[1]['is_deleted'] == case['is_deleted']
 
+    @mock.patch('website.project.tasks.settings.SHARE_URL', 'https://share.osf.io')
+    @mock.patch('website.project.tasks.settings.SHARE_API_TOKEN', 'Token')
+    @mock.patch('website.project.tasks.requests')
+    def test_update_collection_elasticsearch_deleted(self, requests, node_in_collection, collection, user, request_context):
+        now = datetime.datetime.now()
+
+        node_in_collection.deleted = now
```

Comment:
Can you make sure that the `node_in_collection.is_public = True` before running `on_node_updated`? I see you're doing that below, and I'm not convinced that the delete logic will run properly if the node is public. It also looks like from your fixture that it's not made public ever.